keras_nlp/src/layers/modeling/reversible_embedding.py

```diff
@@ -226,10 +234,15 @@ def _int8_call(self, inputs, reverse=False):
 
         return super()._int8_call(inputs)
 
-    def quantize(self, mode):
+    def quantize(self, mode, type_check=True):
         import gc
 
-        if type(self) is not ReversibleEmbedding:
+        if parse(keras.version()) < parse("3.4.0"):
```

Comment:
This is in a lot of lot of places. Maybe move this into `keras_utils.py`?

```python
# This method
def quantize(self, mode, type_check=True):
    assert_quantization_support()
    ...

# Test usage.
if not has_quantization_support():
    self.skipTest("This version of Keras does not support quantization.")
```

Reply:
Using `assert_quantization_support` and `has_quantization_support` is more maintainable. Thanks!
Updated.